presto-cache/src/main/java/com/facebook/presto/cache/filemerge/FileMergeCacheManager.java

```diff
@@ -151,27 +167,69 @@ public void destroy()
     }
 
     @Override
-    public boolean get(FileReadRequest request, byte[] buffer, int offset)
+    public CacheResult get(FileReadRequest request, byte[] buffer, int offset, CacheQuota cacheQuota)
     {
         boolean result = read(request, buffer, offset);
+
+        if (!result && ifExceedQuota(cacheQuota, request)) {
```

Comment:
I think we discussed this. Didn't we think it would be better if we calculate these during put operation?

Reply:
Yes, I thought we had an agreement back then? because in read method, we `always hint the guava Cache`, which is by the original design of our cache system.

That's why we have to check in the read operation if it exceeds quota to decide if we really want to hint the cache

And if we have checked in the read, checking again in put would be excessive

Reply:
Sounds good.